todos/src/utils.js

```diff
@@ -0,0 +1,17 @@
+import store from "./store";
+import { setLoadingState } from "actions";
+
+export const debounce = (func) => {
+  let timer;
+
+  return function (...args) {
+    const context = this;
+    store.dispatch(setLoadingState(true));
```

Comment:
Can we remove the dispatch from here? We want this function to be as generic as possible. 

We can add the loading function in the context of where debounce is used.